### DISCUSSION TITLE
How to create a new problem notification configuration

### DISCUSSION THREAD

**Discussion by ngetichnicholas** [Dynatrace/dynatrace-configuration-as-code#891] [Q&A] [answered]
Hello team, I tried to create a problem notification and alerting profile, but I got an error whereby alerting profile is required to create a notification. How do we handle this with Monaco where we don't have existing configs
![Screenshot from 2023-03-15 10-00-15](https://user-images.githubusercontent.com/79503377/225231342-a093e895-e90c-4c80-b875-7a6aa11daf10.png)

**Comment by UnseenWizzard** [comment, marked as answer]
Hi @ngetichnicholas. 
Problem notifications do always require an alerting profile that they are triggered on. 

To handle this with monaco, you will need to define both the alerting-profile and notification using monaco and reference[0] the profile's ID as a variable to the notification. 

You can find a sample of this (using monaco 2.0 and Settings) here: https://github.com/Dynatrace/dynatrace-configuration-as-code-samples/tree/main/problem_notification_sample (focus mostly on the notification configuration in that sample)

[0] https://www.dynatrace.com/support/help/manage/configuration-as-code/configuration/yaml-configuration#reference

**Reply by ngetichnicholas** [reply]
Thanks @UnseenWizzard , let me try that and let you know

**Reply by ngetichnicholas** [reply]
Hello @UnseenWizzard , this works perfectly

**Comment by andycarlsonbt** [comment]
Is it possible to create an alerting profile and a (related) notification in a single project? It seems that the notification JSON can't 'know' the id of the profile... because it doesn't exist yet. I've tried referencing by name (using 2.0,0-rc8) but that gave me an error. I tried both using the API style JSON and the settings style JSON but got an error both ways. I'd like to create multiple pairs of profiles and notifications so that individual support teams can then tailor the profiles to their own needs.

**Reply by UnseenWizzard** [reply]
Yes, that is possible - by referencing just as you tried already. 
What error did you get?

Did you maybe make a mistake defining the references? 
See [the docs on reference parameters](https://www.dynatrace.com/support/help/manage/configuration-as-code/configuration/yaml-configuration#reference) and [this sample setting up a tag, alerting profile and notification using Settings](https://github.com/Dynatrace/dynatrace-configuration-as-code-samples/tree/main/problem_notification_sample)

**Reply by andycarlsonbt** [reply]
Thanks - I have it working now.

FWIW I was trying to use the profile name in the JSON... this approach works OK for management zones but it seems it doesn't work for alerting profiles. I ran it the wrong way again since you asked. The error looks like...

2023/03/28 10:50:22 INFO        Deploying config project:builtin:problem.notifications:My-Problem-Notification
2023/03/28 10:50:22 WARN  Failed to upsert config "My-Problem-Notification". Waiting for 5s before retrying...
2023/03/28 10:50:27 WARN  Failed to upsert config "My-Problem-Notification". Waiting for 5s before retrying...
2023/03/28 10:50:32 WARN  Failed to upsert config "My-Problem-Notification". Waiting for 5s before retrying...
2023/03/28 10:50:37 ERROR === General Errors ===
2023/03/28 10:50:37 ERROR failed to deploy config project:builtin:problem.notifications:My-Problem-Notification: failed to upsert dynatrace obj: failed to upsert config "My-Problem-Notification" after 3 retries: (HTTP 400)!
    Response was: [{"code":400,"error":{"code":400,"message":"Validation failed for 1 Validators.","constraintViolations":[{"path":"builtin:problem.notifications/0/alertingProfile","message":"Given property 'alertingProfile' with value: 'My Alert Profile Name' violates the following constraint: Must be of type setting.","parameterLocation":"PAYLOAD_BODY","location":null}]},"invalidValue":{"alertingProfile":"My Alert Profile Name","displayName":"SplunkSlurper-to-andy","emailNotification":{"bccRecipients":[],"body":"{ProblemDetailsHTML}","ccRecipients":[],"notifyClosedProblems":true,"recipients":["my.email@myco.com"],"subject":"{State} Problem {ProblemID}: {ImpactedEntity}"},"enabled":true,"type":"EMAIL"}}]
